### PR TITLE
add mc admin monitor command

### PR DESCRIPTION
**Pull request by wlan0**
This is a follow up to the server features added in the PRs below: 
https://github.com/minio/minio/pull/7038
https://github.com/minio/minio/pull/7136

This is a new feature as a part of the effort documented here: https://github.com/minio/mc/issues/2606#issuecomment-452455738

This can be tested by the following:
```bash
// build minio with the server changes for CPU and Mem stats
$ cd $GOPATH/src/github.com/minio/minio
$ go build 

// Create a minio docker image manifest from above build
$ cat > Dockerfile.test << EOF
FROM ubuntu:18.04
COPY minio /usr/bin/
ENV MINIO_ACCESS_KEY minio
ENV MINIO_SECRET_KEY minio123

ENTRYPOINT ["/usr/bin/minio"]
EOF 

// build the docker image
$ docker build -t minio -f Dockerfile.test .

// Clean up all existing docker containers
$ docker rm -fv $(docker ps -qa)

// Setup a minio cluster
$ cat > setupcluster.sh << EOF
#!/bin/sh                                                                                                                                                                            
                                                                                                                                                                                     
minio1=172.17.0.2                                                                                                                                                                    
minio2=172.17.0.3                                                                                                                                                                    
minio3=172.17.0.4                                                                                                                                                                    
minio4=172.17.0.5                                                                                                                                                                    
                                                                                                                                                                                     
docker run  -d minio server http://$minio1:9000/data http://$minio2:9000/data http://$minio3:9000/data http://$minio4:9000/data --address $minio1:9000                               
docker run  -d minio server http://$minio1:9000/data http://$minio2:9000/data http://$minio3:9000/data http://$minio4:9000/data --address $minio2:9000                               
docker run  -d minio server http://$minio1:9000/data http://$minio2:9000/data http://$minio3:9000/data http://$minio4:9000/data --address $minio3:9000                               
docker run  -d minio server http://$minio1:9000/data http://$minio2:9000/data http://$minio3:9000/data http://$minio4:9000/data --address $minio4:9000       
EOF

// run the minio cluster
$ chmod a+x setupcluster.sh && ./setupcluster.sh

// build the mc binary with the changes in this PR
$ cd $GOPATH/src/github.com/minio/mc && go build

// create a mc container (to access the minio cluster inside the docker network)
$ cat > Dockerfile.test << EOF
FROM ubuntu:18.04
COPY mc /usr/bin/


ENTRYPOINT ["/usr/bin/mc"]
EOF

// build the mc conatiner
$ docker build -t mc -f Dockerfile.test .

// run and login to the mc container
$ docker run --entrypoint /bin/bash -ti mc
root@mc-container$ mc config host add minio-docker http://172.17.0.2:9000 minio minio123
mc: Configuration written to `/root/.mc/config.json`. Please update your access credentials.                                                                                         
mc: Successfully created `/root/.mc/share`.                                                                                                                                          
mc: Initialized share uploads `/root/.mc/share/uploads.json` file.                                                                                                                   
mc: Initialized share downloads `/root/.mc/share/downloads.json` file.                                                                                                               
Added `minio-docker` successfully. 
root@mc-container$ mc admin  monitor minio-docker
```
![screenshot 1](https://user-images.githubusercontent.com/1050521/51647804-0c336a00-1f33-11e9-830f-0cfe4e452242.png)



NOTES:
 - [x] Output style verified by AB
 - [x] Follows existing output color scheme for success, warn and error outputs
 - [x] Server changes merged
 - [x] Vendor folder updated with server changes

### CODE REVIEW

**Comment by wlan0** [comment]
NOTE: I ran `govendor remove +u` to prune unused dependencies, and ended up a lot of file changes.

**Comment by nitisht** [comment]
> NOTE: I ran govendor remove +u to prune unused dependencies, and ended up a lot of file changes.

Could you send the govendor prune changes as a separate PR?

**Comment by wlan0** [comment]
@nitisht 

Dependency management for adding minio/minio/pkg/madmin was getting convoluted. It seemingly led to an unending set of dependencies. I updated to go1.11 to simplify this dependency management.

`go.mod` and `go.sum` have been added as a separate commit to this PR

**Comment by wlan0** [comment]
The build failure is a flake - TLS handshake timeout. 

I can't reproduce it locally. I think we should merge this anyways.

File: cmd/admin-monitor.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	// Fetch info of all CPU loads (all minio server instances)
+	cpuMonitors, e := client.ServerCPULoadInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately if error encountered
+		return nil
+	}
+
+	memUsages, e := client.ServerMemUsageInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately id error encountered
```
Comment (nitisht) [inline cmd/admin-monitor.go]: if* error encountered

**Comment by wlan0** [comment]
@nitisht took care of the bug, and addressed all other comments. Please review

**Pull request merged**